__tests__/app.test.js

```diff
@@ -56,3 +56,66 @@ describe("GET *", () => {
       });
   });
 });
+
+describe("GET /api", () => {
+  const endpoints = app._router.stack
+    .filter((r) => r.route)
+    .map((r) => {
+      const route = r.route;
+      const methods = [];
+      if (route.path !== "*") {
+        for (let value of route.stack) {
+          methods.push(`${value.method.toUpperCase()} ${route.path}`);
+        }
+      }
+      return methods;
+    })
+    .flat();
+
+  test("should return 200 status code", () => {
+    return request(app).get("/api").expect(200);
+  });
+
+  test("should return an object on body.endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        expect(typeof body.endpoints).toBe("object");
+      });
+  });
+
+  test("should return an object for currently available endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        endpoints.forEach((endpoint) => {
+          expect(typeof body.endpoints[endpoint]).toBe("object");
+        });
+      });
+  });
+
+  test("each available endpoint should have object with desc., queries, exampleResponse and exampleRequest(if needed)", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        endpoints.forEach((endpoint) => {
+          const endpointObj = body.endpoints[endpoint];
+          if (endpoint === "GET /api") {
+            expect(typeof endpointObj.description).toBe("string");
+          } else if (endpoint.includes("GET") || endpoint.includes("DELETE")) {
+            expect(typeof endpointObj.description).toBe("string");
+            expect(Array.isArray(endpointObj.queries)).toBe(true);
+            expect(typeof endpointObj.exampleResponse).toBe("object");
+          } else {
+            expect(typeof endpointObj.description).toBe("string");
+            expect(Array.isArray(endpointObj.queries)).toBe(true);
+            expect(typeof endpointObj.exampleResponse).toBe("object");
+            expect(typeof endpointObj.exampleRequest).toBe("object");
+          }
+        });
+      });
+  });
+});
```

Comment:
Nice work here 🙂 however the number of tests here is unnecessary. This last one is nice but you could just require the endpoints file in and expect it to look like that. Making is dynamic 👍 Please remove the first 3 tests and i'll leave it up to you to decide if you'd like to refactor the last one